core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -326,12 +326,14 @@ class Partition(val topic: String,
 
   /**
    *  Make the local replica the follower by setting the new leader and ISR to empty
-   *  If the leader replica id does not change, return false to indicate the replica manager
+   *  If the leader replica id does not change and the new epooch is equal or one
+   *  greater (that is, no updates have been missed), return false to indicate the replica manager
    */
   def makeFollower(controllerId: Int, partitionStateInfo: LeaderAndIsrRequest.PartitionState, correlationId: Int): Boolean = {
     inWriteLock(leaderIsrUpdateLock) {
       val newAssignedReplicas = partitionStateInfo.basePartitionState.replicas.asScala.map(_.toInt)
       val newLeaderBrokerId: Int = partitionStateInfo.basePartitionState.leader
+      val oldLeaderEpoch: Int = leaderEpoch
```

Comment:
nit: we typically skip the type for local variables and rely on type inference. Could probably fix this above too.